component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -362,24 +367,52 @@ private boolean isValidSignature(String clientId, SignedJWT signedJWT, String te
                                      String alias) throws OAuthClientAuthnException {
 
         X509Certificate cert = null;
+        String jwksUri = "";
+        boolean isValidSignature = false;
         try {
             cert = (X509Certificate) OAuth2Util.getX509CertOfOAuthApp(clientId, tenantDomain);
         } catch (IdentityOAuth2Exception e) {
             if (log.isDebugEnabled()) {
-                log.debug(e.getMessage(), e);
+                String message = "Unable to retrieve the certificate for the service provider";
+                log.debug(message, e);
+            }
+        }
+        //if cert is null check whether a jwks endpoint is configured for the service provider
+        if (cert == null) {
+            try {
+                ServiceProviderProperty[] spProperties = OAuth2Util.getServiceProvider(clientId).getSpProperties();
+                for (ServiceProviderProperty spProperty : spProperties) {
+                    if (Constants.JWKS_URI.equals(spProperty.getName())) {
+                        jwksUri = spProperty.getValue();
```

Comment:
We might be able to break the loop once required parameter is found. WDYT ?

Reply:
done